src/components/MainContent/index.js

```diff
@@ -23,11 +26,10 @@ function MainContent() {
     };
 
     const onFavoriteClick = () => {
-        // on click look up local storage item
-        // add new coin id to local storage item
-        const newFavorites = [...favorites, data.id];
-        localStorage.setItem("favoriteCoinIds", newFavorites);
-        dispatch(updateFavorites(newFavorites));
+        // dispatch addNewFavorite with argument of coin id saved in state in this component - powering the value of the input
+        dispatch(addNewFavorite(searchedCoin));
+        dispatch(importSavedFavorites());
```

Comment:
Why are you importing saved favorites right after adding a new favorite? double the work for the same end result